zeebe/engine/src/test/java/io/camunda/zeebe/engine/processing/processinstance/MigrateProcessInstanceTest.java

```diff
@@ -1091,6 +1094,164 @@ public void shouldUnsubscribeUnmappedMessageBoundaryEvent() {
         .isTrue();
   }
 
+  @Test
+  public void shouldWriteMigratedEventWhenActiveElementHasAJobIncident() {
+    // given
+    final String sourceProcessId = "process";
+    final String targetProcessId = "process2";
+    final var deployment =
+        ENGINE
+            .deployment()
+            .withXmlResource(
+                Bpmn.createExecutableProcess(sourceProcessId)
+                    .startEvent()
+                    .serviceTask("A", t -> t.zeebeJobType("jobTypeA"))
+                    .endEvent()
+                    .done())
+            .withXmlResource(
+                Bpmn.createExecutableProcess(targetProcessId)
+                    .startEvent()
+                    .serviceTask("B", t -> t.zeebeJobType("jobTypeB"))
+                    .endEvent("target_process_end")
+                    .done())
+            .deploy();
+    final long processInstanceKey =
+        ENGINE.processInstance().ofBpmnProcessId(sourceProcessId).create();
+
+    ENGINE.jobs().withType("jobTypeA").withMaxJobsToActivate(1).activate();
+    RecordingExporter.jobRecords(JobIntent.CREATED).withType("jobTypeA").await();
+
+    final Record<JobRecordValue> failedEvent =
+        ENGINE.job().withType("jobTypeA").ofInstance(processInstanceKey).withRetries(0).fail();
+
+    final Record<IncidentRecordValue> incident =
+        RecordingExporter.incidentRecords(IncidentIntent.CREATED)
+            .withJobKey(failedEvent.getKey())
+            .getFirst();
+
+    final long targetProcessDefinitionKey =
+        extractProcessDefinitionKeyByProcessId(deployment, targetProcessId);
+
+    // when
+    ENGINE
+        .processInstance()
+        .withInstanceKey(processInstanceKey)
+        .migration()
+        .withTargetProcessDefinitionKey(targetProcessDefinitionKey)
+        .addMappingInstruction("A", "B")
+        .migrate();
+
+    // then
+    assertThat(
+            RecordingExporter.incidentRecords(IncidentIntent.MIGRATED)
+                .withProcessInstanceKey(processInstanceKey)
+                .getFirst()
+                .getValue())
+        .describedAs("Expect that process definition is updated")
+        .hasProcessDefinitionKey(targetProcessDefinitionKey)
+        .hasBpmnProcessId(targetProcessId)
+        .describedAs("Expect that element id changed due to mapping")
+        .hasElementId("B");
```

Comment:
❌ We must ensure that only these three properties change. The rest should stay the same.

For example:
```java
.hasJobKey(incident.getValue().getJobKey())
```